zebra-state/src/service.rs

```diff
@@ -270,6 +270,105 @@ impl StateService {
             state: IterState::NonFinalized(hash),
         }
     }
+
+    /// Return a list of block hashes in the best chain, following the first matching hash in `known_blocks`.
+    ///
+    /// Starts from the first matching hash in the best chain, ignoring all other hashes in `known_blocks`.
```

Comment:
Sorry, this requirement isn't documented in the bitcoin wiki or Zcash spec, so I missed it:
```suggestion
    /// Starts from the first matching hash in the best chain, ignoring all other hashes in `known_blocks`.
    /// If there is no matching hash in the best chain, starts from the genesis block.
    ///
```